master/internal/rm/kubernetesrm/pods.go

```diff
@@ -1233,6 +1277,18 @@ func (p *pods) handleGetAgentsRequest() *apiv1.GetAgentsResponse {
 	return response
 }
 
+func (p *pods) handleGetAgentRequest(agentID string) *apiv1.GetAgentResponse {
+	nodeSummaries := p.summarizeClusterByNodes()
+	_, nodesToPools := p.getNodeResourcePoolMapping(nodeSummaries)
+	agentSummary, ok := nodeSummaries[agentID]
+	if !ok {
+		p.syslog.Warnf("No agent with id %s", agentID)
```

Comment:
we should todo returning an error here. it's just annoying to change the interface since it is also in EE.

Reply:
made a ticket for this and added the todo!